synapse/models/proj.py

```diff
@@ -0,0 +1,160 @@
+import synapse.common as s_common
+
+import synapse.lib.module as s_module
+
+prioenums = (
+    (0, 'none'),
+    (10, 'lowest'),
+    (20, 'low'),
+    (30, 'medium'),
+    (40, 'high'),
+    (50, 'highest'),
+)
+
+statusenums = (
+    (0, 'new'),
+    (10, 'in validation'),
+    (20, 'in backlog'),
+    (30, 'in sprint'),
+    (40, 'in progress'),
+    (50, 'in review'),
+    (60, 'completed'),
+    (70, 'done'),
+)
+
+class ProjectModule(s_module.CoreModule):
+
+    async def initCoreModule(self):
+        self.model.form('proj:project').onAdd(self._onAddProj)
+        self.model.form('proj:project').onDel(self._onDelProj)
+
+    async def _onAddProj(self, node):
+        # ref counts on authgates?
+        gateiden = node.ndef[1]
+
+        await self.core.auth.addAuthGate(node.ndef[1], 'storm:project')
+
+        useriden = node.snap.user.iden
+
+        rule = (True, ('project', 'admin'))
+        await node.snap.user.addRule(rule, gateiden=gateiden)
+
+    async def _onDelProj(self, node):
+        gateiden = node.ndef[1]
+        await self.core.auth.delAuthGate(gateiden)
+
+    def getModelDefs(self):
+        return (
+
+            ('proj', {
+
+                'types': (
+                    ('proj:epic', ('guid', {}), {}),
+                    ('proj:ticket', ('guid', {}), {}),
+                    ('proj:sprint', ('guid', {}), {}),
+                    ('proj:comment', ('guid', {}), {}),
+                    ('proj:project', ('guid', {}), {}),
+                ),
+
+                'forms': (
```

Comment:
Looks good based on initial discussions. Just confirm before we commit that the structure (and especially naming conventions) are generic enough form any type of project/tasking and we're not confining ourselves too narrowly to sprint-style planning.

Reply:
Yep, i'll def be eventually constructing "backlogs" ( which are really just ordered todo lists ) and the entire "sprint" thing will essentially be an optional element to project planning.